### PR TITLE
Adds Rails 7 Support

### PR DESCRIPTION
**Pull request by nikz**
* Swaps to `Appraisal` gem for using different gemsets when testing
* Modify JS exception tracking injection to handle Rack::Body
* Adds Rails 7.1 dummy app and add to CI matrix

### CODE REVIEW

File: spec/rails_applications/7.1.3/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+!/log/.keep
+!/tmp/.keep
+
+# Ignore pidfiles, but keep the directory.
+/tmp/pids/*
+!/tmp/pids/.keep
+
+# Ignore storage (uploaded files in development and any SQLite databases).
+/storage/*
+!/storage/.keep
```
Comment (sumitramanga) [inline spec/rails_applications/7.1.3/.dockerignore:29]: You've mentioned above what you're keeping but not for all sections. `!/storage/.keep` means you're keeping this directory correct? Would be good to add in the comment in that case plus for the others :)

File: spec/rails_applications/7.1.3/README.md
Comment (sumitramanga) [inline spec/rails_applications/7.1.3/README.md:1]: Should this be filled in now or later down the line?

File: spec/rails_applications/7.1.3/app/assets/images/.keep
Comment (sumitramanga) [inline spec/rails_applications/7.1.3/app/assets/images/.keep:1]: Is this intentional?

File: spec/rails_applications/7.1.3/app/mailers/application_mailer.rb
Comment (sumitramanga) [inline spec/rails_applications/7.1.3/app/mailers/application_mailer.rb:1]: Just trying to figure out if this is needed

File: spec/rails_applications/7.1.3/.dockerignore (hunk 1 above)
Comment (nikz) [reply spec/rails_applications/7.1.3/.dockerignore:29]: Hi @sumitramanga this is just part of the standard rails layout - so I wouldn't want to make any changes here as this is upstream from Rails (see https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/dockerignore.tt#L30)

File: spec/rails_applications/7.1.3/README.md
Comment (nikz) [reply spec/rails_applications/7.1.3/README.md:1]: Same as previous - I would ignore this as part of the Rails skeleton.

File: spec/rails_applications/7.1.3/config/credentials.yml.enc [hunk 2]
```diff
@@ -0,0 +1 @@
+v+qpJS9dHtYdjpHPCRVcjajq+Tlb6ONmeul+eI9CFyFtJQMcejkgV8w8H0w90kwselljjrjD+YxZUq8zarwhrmWC+03M4JOwJGDE5C/Iaofa2pT5qtjZSP7/wL9v4G6/lFpKBvT+teCRhXs+6uvcWACyFotEKZq89v6ApMBgzF94sioz5LmY51gFSIk1EhaMuEut0lUWOYn/mzBRA/sdYymEM4ynOg5AglQ2Q1c34JIQtrWnGrtwMD9C/XnQ96vUZcQd4wUQUjYtF0xnJNx/MknaHMdbrS26zXMpmBveh/O8a67x/ozO/c8hjnYeMPMZ0lOdHnbyWsrdpCjhHkKA7foc3KuCkMEsHXvFbWnlip7doOaF4tKwTq9z79gxshYJTzQ2CXrIgDignUcNlecVj3Krkmpb--A0mHiuQO+RWqb84v--VvD6hXbRbSQZ5vYc2YVqjw==
```
Comment (sumitramanga) [inline spec/rails_applications/7.1.3/config/credentials.yml.enc:1]: Do we want to be committing this? Seems sensitive

Comment (nikz) [reply spec/rails_applications/7.1.3/config/credentials.yml.enc:1]: Encrypted: https://edgeguides.rubyonrails.org/security.html#custom-credentials

Inside there it's just a random `secret_key_base` and a few comments

File: spec/rails_applications/7.1.3/app/mailers/application_mailer.rb
Comment (sumitramanga) [reply spec/rails_applications/7.1.3/app/mailers/application_mailer.rb:1]: Ignore - After getting some knowledge share, this is just a file required from the standard rails app generator

File: spec/rails_applications/7.1.3/app/assets/images/.keep
Comment (sumitramanga) [reply spec/rails_applications/7.1.3/app/assets/images/.keep:1]: Ignore - After getting some knowledge share, this is just a file required from the standard rails app generator

File: spec/rails_applications/7.1.3/.dockerignore (hunk 1 above)
Comment (sumitramanga) [reply spec/rails_applications/7.1.3/.dockerignore:29]: No worries! Thank you for the context

**Review by sumitramanga — COMMENTED** [review]
I'm happy with this PR, great work! Just need to check with Panos next steps before this gets released to the public :) I'll let you know how I get on with this

**Review by sumitramanga — APPROVED** [review]

**Pull request merged**